src/ray/gcs/tables.h

```diff
@@ -354,6 +367,21 @@ class TaskLeaseTable : public Table<TaskID, TaskLeaseData> {
     pubsub_channel_ = TablePubsub::TASK_LEASE;
     prefix_ = TablePrefix::TASK_LEASE;
   }
+
+  Status Add(const JobID &job_id, const TaskID &id, std::shared_ptr<TaskLeaseDataT> &data,
+             const WriteCallback &done) override {
+    RAY_RETURN_NOT_OK((Table<TaskID, TaskLeaseData>::Add(job_id, id, data, done)));
```

Comment:
Thanks!

Reply:
Actually, I need the parentheses for it to build for some reason...I don't know exactly why, but the compiler seems to think that there are 2 arguments to the macro, maybe because of the templated argument?

Reply:
Is the reason that you have to add `Table<TaskID, TaskLeaseData>::` because the two methods have the same name? Would that go away if we just change this method name from `Add` to something else like `UpdateTaskLease`?

Reply:
Just googled this, the reason is because the macro processor doesn't know about template, thus recognizes `Table<TaskID` as a parameter.
Someone suggests this way:
```c++
#define COMMA ,
RAY_RETURN_NOT_OK(Table<TaskID COMMA TaskLeaseData>::(...))
```
It works. But I think adding an extra pair of parentheses is easier. Or as @robertnishihara suggested, use another function name.

Reply:
Sounds good!

Reply:
I decided against changing the function name since I believe it's nicer to have the `TaskLeaseTable` still implement the `Table<TaskID, TaskLeaseData>` interface.